api/views/course.py

```diff
@@ -182,12 +182,33 @@ def course_grade_book(self, request, pk):
 
         teams = []
         for event in events.all():
-            teams.extend(event.team_set.all())
+            if request.user.is_student:
+                teams.extend(event.team_set.filter(course_registrations=request.user.id))
+            else:
+                teams.extend(event.team_set.all())
 
         results = []
         for team in teams:
-            for course_reg in team.course_registrations.filter(registration_type="STUDENT", status="VERIFIED"):
+            if request.user.is_student:
+                course_registrations = team.course_registrations.filter(
+                    registration_type="STUDENT", status="VERIFIED", user=request.user.id
+                )
+            else:
+                course_registrations = team.course_registrations.filter(registration_type="STUDENT", status="VERIFIED")
+            for course_reg in course_registrations:
+                question_details = []
                 consent = course_reg.user.consents.last()
+                questions = team.event.question_set.all().order_by("title")
+                for question in questions:
+                    submissions = question.user_junctions.get(user=course_reg.user).submissions.all()
+                    best_attempt_grade = max([submission.grade for submission in submissions]) if submissions else 0
```

Comment:
If the logic of best attempt changes in the future this part will break and we will have no idea we have to fix it here until someone report the issue. It's better to use uqj's here too.